icu4c/source/i18n/number_longnames.cpp

```diff
@@ -431,13 +431,20 @@ void getMeasureData(const Locale &locale,
     subKey.append(unit.getType(), status);
     subKey.append("/", status);
 
+    // TODO(ICU-21938): remove after fixing the data in CLDR-15421
+    const char *unitSubType = unit.getSubtype();
```

Comment:
CLDR-15241 is not going to be fixed, CLDR is behaving as intended, see comments in that ticket. Any fix here needs to be a real fix, not a hack.